include/eve/detail/function/simd/wasm/make.hpp

```diff
@@ -0,0 +1,120 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/fundamental_cardinal.hpp>
+#include <eve/as.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/detail/alias.hpp>
+#include <eve/detail/category.hpp>
+#include <eve/detail/spy.hpp>
+
+namespace eve::detail
+{
+//================================================================================================
+// enumerated make - 128bits
+//================================================================================================
+template<real_scalar_value T, typename S, typename... Vs>
+EVE_FORCEINLINE auto
+make(eve::as<wide<T, S>>, Vs... vs) noexcept requires std::same_as<abi_t<T, S>, wasm_128_>
+{
+  static_assert(sizeof...(Vs) <= S::value, "[eve::make] - Invalid number of arguments");
+
+  constexpr auto c = categorize<wide<T, S>>();
+  using enum category;
+
+  if constexpr( S::value == 2 )
+  {
+    if constexpr( c == float64x2 ) return wasm_f64x2_make(vs...);
+    if constexpr( c == int64x2 ) return wasm_i64x2_make(vs...);
+    if constexpr( c == uint64x2 ) return wasm_u64x2_make(vs...);
+    if constexpr( c == float32x4 ) return wasm_f32x4_make(vs..., 0.f, 0.f);
+    if constexpr( c == int32x4 ) return wasm_i32x4_make(vs..., 0, 0);
+    if constexpr( c == uint32x4 ) return wasm_u32x4_make(vs..., 0, 0);
+    if constexpr( c == int16x8 ) return wasm_i16x8_make(vs..., 0, 0, 0, 0, 0, 0);
+    if constexpr( c == uint16x8 ) return wasm_u16x8_make(vs..., 0, 0, 0, 0, 0, 0);
```

Comment:
if you wrap these `u16x8_make` into

```
namespace wsm {
inline auto wasm_u32x4_make(std::uint32_t v0,  std::uint32_t v1, std::uint32_t v2, std::uint32_t v3) {
   return wasm_u32x4_make(v0, v1, v2, v3);
} 
}
```

You will be able to use variadics here